### DISCUSSION TITLE
Bun v1.0.17

### DISCUSSION THREAD

**Discussion by Jarred-Sumner** [oven-sh/bun#7596] [Announcements]
To install Bun v1.0.17:
```bash
curl https://bun.sh/install | bash
# or you can use npm
# npm install -g bun
```

To upgrade to Bun v1.0.17:
```bash
bun upgrade
```

**[Read Bun v1.0.17's release notes on Bun's blog](https://bun.sh/blog/bun-v1.0.17)**

## Thanks to the 6 contributors who made this release possible:

- [@DaleSeo](https://github.com/DaleSeo)
- [@dylan-conway](https://github.com/dylan-conway)
- [@hustLer2k](https://github.com/hustLer2k)
- [@Jarred-Sumner](https://github.com/Jarred-Sumner)
- [@NReilingh](https://github.com/NReilingh)
- [@paperdave](https://github.com/paperdave)

<hr /><em>This discussion was created from the release <a href='https://github.com/oven-sh/bun/releases/tag/bun-v1.0.17'>Bun v1.0.17</a>.</em>

**Comment by MitchTalmadge** [comment]
Incredible!! Thank you everyone for all your hard work!

**Comment by zhendalf** [comment]
In my project, upgrading to bun `1.0.17` increased the time of `bun install` in CI from ~7s to ~120s. It appears that the difference is caused by all the scripts that now run on `postinstall`. 

I recognize that running scripts solved a number of issues, but in my case it didn't solve anything, but added a lot of unnecessary overhead.

According to Bun [docs](https://bun.sh/docs/cli/install#lifecycle-scripts):

> Unlike other npm clients, Bun does not execute arbitrary lifecycle scripts like `postinstall` for installed dependencies. Executing arbitrary scripts represents a potential security risk.

My understanding is that since 1.0.17 this is no longer the case.

I want to keep Bun up to date, but how do I revert to the previous behavior? Installing dependencies in 7s felt like magic, but now the magic is gone 😔. I may need to run the scripts for certain dependencies, and specifying those in `trustedDependencies` doesn't seem like a big deal.

**Reply by zhendalf** [reply]
There is an issue for this: https://github.com/oven-sh/bun/issues/7689
